packages/bitcore-node/src/modules/ethereum/models/transaction.ts

```diff
@@ -0,0 +1,224 @@
+    parentChain?: string;
+    forkHeight?: number;
+    initialSyncComplete: boolean;
+    chain: string;
+    network: string;
+    mempoolTime?: Date;
+  }) {
+    let { blockTimeNormalized, chain, height, network, parentChain, forkHeight } = params;
+    if (parentChain && forkHeight && height < forkHeight) {
+      const parentTxs = await EthTransactionStorage.collection
+        .find({ blockHeight: height, chain: parentChain, network })
+        .toArray();
+      return parentTxs.map(parentTx => {
+        return {
+          updateOne: {
+            filter: { txid: parentTx.txid, chain, network },
+            update: {
+              $set: {
+                ...parentTx,
+                wallets: new Array<ObjectID>()
+              }
+            },
+            upsert: true,
+            forceServerObjectId: true
+          }
+        };
+      });
+    } else {
+      return Promise.all(
+        params.txs.map(async (tx: IEthTransaction) => {
+          const { to, txid, from } = tx;
+          const sentWallets = await WalletAddressStorage.collection.find({ chain, network, address: from }).toArray();
+          const receivedWallets = await WalletAddressStorage.collection.find({ chain, network, address: to }).toArray();
+          const wallets = _.uniqBy(sentWallets.concat(receivedWallets).map(w => w.wallet), w => w.toHexString());
+
+          return {
+            updateOne: {
+              filter: { txid, chain, network },
+              update: {
+                $set: {
+                  ...tx,
+                  blockTimeNormalized,
+                  wallets
+                }
+              },
+              upsert: true,
+              forceServerObjectId: true
+            }
+          };
+        })
+      );
+    }
+  }
+
+  async pruneMempool(params: {
+    txs: Array<IEthTransaction>;
+    height: number;
+    parentChain?: string;
+    forkHeight?: number;
+    chain: string;
+    network: string;
+    initialSyncComplete: boolean;
+  }) {
+    const { chain, network, initialSyncComplete } = params;
+    if (!initialSyncComplete) {
+      return;
+    }
+    let prunedTxs = {};
+    if (Object.keys(prunedTxs).length) {
+      prunedTxs = Object.keys(prunedTxs);
+      await Promise.all([
+        this.collection.update(
+          { chain, network, txid: { $in: prunedTxs } },
+          { $set: { blockHeight: SpentHeightIndicators.conflicting } },
+          { w: 0, j: false, multi: true }
+        )
+      ]);
+    }
+    return;
+  }
+
+  getTransactions(params: { query: any; options: StreamingFindOptions<IEthTransaction> }) {
+    let originalQuery = params.query;
+    const { query, options } = Storage.getFindOptions(this, params.options);
+    const finalQuery = Object.assign({}, originalQuery, query);
+    return this.collection.find(finalQuery, options).addCursorFlag('noCursorTimeout', true);
+  }
+
+  abiDecode(input: string) {
+    try {
+      try {
+        AbiDecoder.addABI(ERC20Abi);
```

Comment:
If we addABI on top of the file, it will always decode properly so we would not know what type of erc the token is. Need to think about this for optimization.